query-engine/core/src/interactive_transactions/transaction.rs

```diff
@@ -197,27 +197,54 @@ impl InteractiveTransaction {
         })
     }
 
-    pub async fn commit(&mut self) -> crate::Result<()> {
-        tx_timeout!(self, "commit", async {
+    pub fn depth(&mut self) -> i32 {
+        match self.state.as_open("depth") {
+            Ok(state) => state.depth(),
+            Err(_) => 0,
```

Comment:
I'd prefer this to return `crate::Result<i32>` and use `tx.depth()?` in the tx manager

Reply:
Ok, I'll look into it

Reply:
Resolved 👍